flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/adaptive/CreatingExecutionGraph.java

```diff
@@ -143,7 +143,7 @@ private void handleExecutionGraphCreation(
             } else {
                 logger.debug(
                         "Failed to reserve and assign the required slots. Waiting for new resources.");
-                context.goToWaitingForResources();
+                context.goToWaitingForResources(null);
```

Comment:
🤔 Looking back on Chesnay's comment [1] Is passing null here correct? If not, I'd rather mark it as a bug and follow up in a separate PR to get things moving.

[1] https://github.com/apache/flink/pull/21981#pullrequestreview-1307055034

Reply:
Good point, thanks!
No, it's not correct. But the fix isn't complex - I pushed c4aaddb464f39d454d6d4f362cf4a0cf5b4973af to add the `previousExecutionGraph` to `CreatingExecutionGraph` and then pass it to `WaitingForResources`.